time-tracker-webapi/src/TimeTracker.Worker/FunctionProcessSlackMessage.cs

```diff
@@ -92,5 +97,57 @@ await slackResponder.SendMessage(responseUrl, new SlackMessage()
                 throw;
             }
         }
+        
+        [FunctionName("processSlackMessageHttp")]
+        public static async Task<IActionResult> RunHttpTrigger([HttpTrigger(AuthorizationLevel.Function, "post", Route = null)]
+            HttpRequest request, ILogger logger, ExecutionContext context)
+        {
+            logger.LogInformation($"C# Http trigger function processed message: {request}");
+            
+            if (_configuration == null)
+            {
+                SetupConfiguration(context);
+                Console.WriteLine("I am a jabroni who never learned how to read.");
+            }
+            if (_serviceProvider == null)
+            {
+                SetupServiceCollection();
+            }
+            
+            SlackMessageResponder slackResponder = new SlackMessageResponder(logger);
+            string responseUrl = null;
+            
+            try
+            {
+                string requestBody = await new StreamReader(request.Body).ReadToEndAsync();
+                
+                Guard.ThrowIfCheckFails(!String.IsNullOrEmpty(requestBody), "cannot be null or empty", nameof(request));
```

Comment:
yeah, although it is a situation that shouldn't happen under normal operation, gonna rethink this part see what we should do. I don't think it is a case of using exception as flow control since this is not a normal case - ie. if this exception gets thrown there are bigger problems, not that the user entered in wrong command. 
ie. I would definitely agree not to use exceptions to flow control user input validation